python/cugraph/cugraph/utilities/utils.py

```diff
@@ -205,12 +195,26 @@ def get_traversed_path_list(df, id):
     return answer
 
 
-# FIXME: if G is a Nx type, the weight attribute is assumed to be "weight", if
-# set. An additional optional parameter for the weight attr name when accepting
-# Nx graphs may be needed.  From the Nx docs:
-# |      Many NetworkX algorithms designed for weighted graphs use
-# |      an edge attribute (by default `weight`) to hold a numerical value.
-def ensure_cugraph_obj(obj, nx_weight_attr=None, matrix_graph_type=None):
+def ensure_valid_dtype(input_graph, vertex_pair, func_name):
```

Comment:
An alternative to having the caller pass in `func_name` could be to use `inspect` to find the calling function name:
```python
import inspect
func_name = inspect.stack()[1].function
```
Since calling `inspect` is a bit expensive, only do it if the check fails right before issuing the warning. The downside here is that it assumes the calling function name should be in the message which may be wrong if an API calls a helper which calls this.

Or just continue to require `func_name` be passed in, but this also has downsides (similar problem if an API calls a helper which calls this).